src/Middleware/Diagnostics/src/DeveloperExceptionPage/Views/ErrorPage.cshtml

```diff
@@ -21,312 +21,342 @@
+                    }
+                    @if (!string.IsNullOrEmpty(Model.Endpoint.HttpMethods))
+                    {
                         <tr>
-                            <td>@Resources.ErrorPageHtml_EndpointDisplayName</td>
-                            <td>@Model.Endpoint.DisplayName</td>
+                            <td>@Resources.ErrorPageHtml_EndpointRouteHttpMethod</td>
+                            <td>@Model.Endpoint.HttpMethods</td>
                         </tr>
-                        @if (!string.IsNullOrEmpty(Model.Endpoint.RoutePattern))
-                        {
-                            <tr>
-                                <td>@Resources.ErrorPageHtml_EndpointRoutePattern</td>
-                                <td>@Model.Endpoint.RoutePattern</td>
-                            </tr>
-                        }
-                        @if (Model.Endpoint.Order != null)
-                        {
-                            <tr>
-                                <td>@Resources.ErrorPageHtml_EndpointRouteOrder</td>
-                                <td>@Model.Endpoint.Order</td>
-                            </tr>
-                        }
-                        @if (!string.IsNullOrEmpty(Model.Endpoint.HttpMethods))
-                        {
-                            <tr>
-                                <td>@Resources.ErrorPageHtml_EndpointRouteHttpMethod</td>
-                                <td>@Model.Endpoint.HttpMethods</td>
-                            </tr>
-                        }
-                    </tbody>
-                </table>
-            }
-            else
-            {
-                <p>@Resources.ErrorPageHtml_NoEndpoint</p>
-            }
-            <h2>@Resources.ErrorPageHtml_RouteValues</h2>
-            @if (Model.RouteValues != null && Model.RouteValues.Any())
-            {
-                <table>
-                    <thead>
+                    }
+                </tbody>
+            </table>
+        }
+        else
+        {
+            <p>@Resources.ErrorPageHtml_NoEndpoint</p>
+        }
+        <h2>@Resources.ErrorPageHtml_Metadata</h2>
+        @if (Model.Endpoint?.Metadata.Count > 0)
+        {
+            <table>
+                <thead>
+                    <tr>
+                        <th>@Resources.ErrorPageHtml_MetadataTypeColumn</th>
+                        <th>@Resources.ErrorPageHtml_MetadataDetail</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    @foreach (var metadata in Model.Endpoint.Metadata)
+                    {
                         <tr>
-                            <th>@Resources.ErrorPageHtml_VariableColumn</th>
-                            <th>@Resources.ErrorPageHtml_ValueColumn</th>
+                            <td><span title="@metadata.GetType().FullName">@metadata.GetType().Name</span></td>
+                            <td>@HtmlEncodeAndReplaceLineBreaks(metadata.ToString())</td>
```

Comment:
I include both with `HtmlEncodeAndReplaceLineBreaks` and without in the description. I suggest to keep the one without. The razor would encode the html and omit the new lines.